include/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.hpp

```diff
@@ -105,6 +107,8 @@ class Segment
         /// @return Delta mass
         Mass computeDeltaMass() const;
 
+        Array<Maneuver> generateManeuvers() const;
```

Comment:
Just asking here whether `generate` is the right prefix. Would the be generated on-demand or are they stored beforehand?

Reply:
They would be generated on demand (to not bloat the segmentSolution with an `Array<Maneuver>` memvar that it might not need for coast segments.

Reply:
I think Pau is asking: should we use generate, or rather extract or sth else

Reply:
Yeah extract sounds good too, I'll change it to that